pandas/core/internals/blocks.py

```diff
@@ -591,23 +591,16 @@ def _astype(self, dtype, copy=False, errors='raise', values=None,
 
             return self.make_block(Categorical(self.values, dtype=dtype))
 
-        # convert dtypes if needed
         dtype = pandas_dtype(dtype)
+        if isinstance(dtype, PandasDtype):
+            dtype = dtype.numpy_dtype
+
         # astype processing
         if is_dtype_equal(self.dtype, dtype):
             if copy:
                 return self.copy()
             return self
 
-        klass = None
```

Comment:
I believe that this is unnecessary now... The `.astype` should get us an array of the correct type, and from there we'll get the correct block in `make_block`.